crates/polars-time/src/group_by/dynamic.rs

```diff
@@ -34,6 +34,7 @@ pub struct DynamicGroupOptions {
     pub include_boundaries: bool,
     pub closed_window: ClosedWindow,
     pub start_by: StartBy,
+    pub int_range: bool, // Whether the index column is a range column
```

Comment:
Perhaps this should be private (as we do not want users to set this "flag")

Reply:
No, we don't.

We can add a new `DynamicGroupOptionsIR` where we add private flags set by the IR.